test/e2e/suite/cleanup.go

```diff
@@ -0,0 +1,35 @@
+package suite
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+const deferCleanupTimeout = 5 * time.Minute
+
+type TestCleanup struct {
+	cleanups []func(context.Context)
+}
+
+func (t *TestCleanup) Register(f func(context.Context)) {
```

Comment:
maybe add some doc comments in these two methods to give context on what this does and why is useful?